.github/workflows/ci.yaml

```diff
@@ -0,0 +1,245 @@
+        sudo cmake --install cmake-build-dir --prefix /usr/local
+
+    - name: build qBittorrent
+      run: |
+        cmake -B build -G "Ninja" \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -D${{ matrix.qbt_gui }} \
+          -DVERBOSE_CONFIGURE=ON \
+          --graphviz=build/target_graph.dot
+        cmake --build build
+
+    - name: upload artifact as zip
+      uses: actions/upload-artifact@v2.1.3
+      with:
+        name: qBittorrent-CI-Ubuntu_${{ matrix.os }}-${{ matrix.qbt_gui }}
+        path: |
+          build/compile_commands.json
+          build/target_graph.dot
+          build/qbittorrent
+          build/qbittorrent-nox
+          libtorrent/cmake-build-dir/compile_commands.json
+          libtorrent/cmake-build-dir/target_graph.dot
+
+  ci_windows:
+    name: Windows + vcpkg
+
+    runs-on: windows-2019
+
+    defaults:
+      run:
+        shell: pwsh
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v2.3.2
+
+    # - ninja is needed for building qBittorrent (because it's preferrable, not a hard requirement)
+    - name: install additional required packages with chocolatey
+      run: |
+        choco install ninja
+
+    - name: setup vcpkg (cached, if possible)
+      uses: lukka/run-vcpkg@v3.3
+      with:
+        vcpkgDirectory: ${{ env.VCPKG_DEST_WIN }}
+        vcpkgGitCommitId: ${{ env.VCPKG_COMMIT }}
+        setupOnly: true
+
+    # Tell vcpkg to only build Release variants of the dependencies
+    - name: configure vcpkg triplet overlay for release builds only
+      run: |
+        New-Item -Path ${{ github.workspace }} -Name "triplets_overlay" -ItemType Directory
+        Copy-Item ${{ env.RUNVCPKG_VCPKG_ROOT }}/triplets/x64-windows-static.cmake `
+          ${{ github.workspace }}/triplets_overlay/x64-windows-static-release.cmake
+        Add-Content ${{ github.workspace }}/triplets_overlay/x64-windows-static-release.cmake `
+          -Value "set(VCPKG_BUILD_TYPE release)"
+
+    # clear buildtrees after each package installation to reduce disk space requirements
+    - name: install dependencies via vcpkg
+      run: |
+        $packages = `
+          "boost-circular-buffer:x64-windows-static-release",
+          "libtorrent:x64-windows-static-release",
+          "qt5-base:x64-windows-static-release",
+          "qt5-svg:x64-windows-static-release",
+          "qt5-tools:x64-windows-static-release",
+          "qt5-winextras:x64-windows-static-release"
+        foreach($package in $packages)
+        {
+          ${{ env.RUNVCPKG_VCPKG_ROOT }}/vcpkg.exe install $package `
+            --overlay-triplets=${{ github.workspace }}/triplets_overlay `
+            --clean-after-build
+        }
+
+    # NOTE: this is necessary to correctly find and use cl.exe with the Ninja generator for now
+    - name: setup devcmd
+      uses: ilammy/msvc-dev-cmd@v1.3.0
+      with:
+        arch: x64
+        spectre: true
```

Comment:
I think we should keep the builds simple by omitting this setting. Who knows what (unexpected) impact would this setting bring in?

Reply:
@Chocobo1 Fixed. Also removed `arch`, since `x64` is the default anyway.